demos/mlsys_2020/lowering_demo.ipynb

```diff
@@ -0,0 +1,228 @@
+{
```

Comment:
Is there a way to move the majority of this code into a python file? It's hard to review it in this form.

Reply:
Thanks. I kept the exact same structure as the other demo, in case we want to merge both. 
But yes, stuff can definitely be moved around. Let me address this in another PR

Reply:
Would it be possible to do it in this PR or would that make a mess of things?

I only ask because I'm finding it hard to review the code.

Reply:
I agree it is hard to review.
Although, it is the same code that was in Denise's PR, minus the grids for displaying the inputs and outputs, so you might have already gone through it.
My reasoning behind keeping it the same way is this - if we want to merge the two demos(like she had asked me to) , we can restructure them together. If we keep them separate, then we can work on making them modular separately. 
Additionally, I think voila only works on the notebook. I am looking into whether it can work on a python file, if not, we probably have to keep all the interactive stuff in the notebook :(

Reply:
To me it seems that whether we merge the two demos or not is orthogonal to the request to split out as much logic as possible from the notebook to make it easier to review. Would it be possible to put this split out logic into a separate python file for now and then merge the two python files later if we decide that's a good idea?

I think you're right that I think voila only works on a notebook, but I don't understand why it can't work with code that is imported from a python file? Basically it seems like the only thing we need in the notebook is something like:

```
import demo
demo.run()
```

Reply:
Makes sense, I will separate the logic into a python file.